Upgrading pipeline, codes and utils for taxi dataset

dev-nid opened the pull request:
- Airflow tutorial for chicago taxi dataset updated.
- Readme updated

Commit:
Skips ranking pipeline tests when struct2tensor is not available.

PiperOrigin-RevId: 464440844

tfx-copybara committed:
Bump up minimum apache-beam version to 2.40.0 to support new run_inference API.

PiperOrigin-RevId: 464616132

chongkong committed:
Add ResolverStrategy support in `input_graph_resolver`.

PiperOrigin-RevId: 464722882

tfx-copybara committed:
Adding support for dynamic execution properties in TFX

PiperOrigin-RevId: 464951410

aayooush committed:
Adding SequenceExample support to TFX Transform Executor

PiperOrigin-RevId: 465159203

tfx-copybara committed:
Create ExampleDiff component in TFlex

PiperOrigin-RevId: 465184092

chongkong committed:
Change ResolverOp.apply signature to use flexible data type.

PiperOrigin-RevId: 465278557

micahjsmith committed:
Add verify_artifacts util, add logic to cache_utils, and fix lint issues

micahjsmith committed:
fix test

micahjsmith committed:
fix lint

micahjsmith committed:
fix typo

tfx-copybara committed:
Include update times in node_run_states_history.

PiperOrigin-RevId: 465404837

micahjsmith committed:
add test for cache_utils.get_cached_outputs

tfx-copybara committed:
Automated rollback of commit 23a809c2bb50759d0347d652dd2b8b5cb1ca7b59

PiperOrigin-RevId: 465488408

sanatmpa1 committed:
Updating broken link on beam_orchestrator.md

Fixing broken link for `TFX example on beam`

micahjsmith committed:
add more test cases for artifact_utils.verify_artifacts

micahjsmith committed:
revert reorder tf import

iindyk committed:
Adding support of outputting parquet files from TFX Transform.

PiperOrigin-RevId: 465626255

mungsoo committed:
Allow JSON value artifacts to be passed in and returned by components as JsonValue artifacts

PiperOrigin-RevId: 465690612

markmcd committed:
Add kwarg support for `embedding_fn`.

The `neural_structured_learning` package passes kwargs to the embedding function, so ensure that its use in the docs/examples includes it.

PiperOrigin-RevId: 465951403

tfx-copybara committed:
Merge pull request #5111 from tensorflow:sanatmpa1-patch-2

PiperOrigin-RevId: 466004570

tfx-copybara committed:
nop

PiperOrigin-RevId: 466112412

tfx-copybara committed:
Updates TFDV skew pipeline to support counting the incidence of each combination (test-value, base-value, feature-name) - analogous to a confusion matrix per feature - for values of matching examples. Initially supports univalent bytes features.

PiperOrigin-RevId: 466194064

hawkinsp committed:
[NumPy] Replace numpy.asscalar(x) with x.item() in preparation for upgrading NumPy to 1.23.

NumPy 1.23 removes numpy.asscalar() (https://numpy.org/doc/stable/release/1.23.0-notes.html#expired-deprecations), which has been deprecated since NumPy 1.16 (https://numpy.org/doc/stable/release/1.16.0-notes.html#new-deprecations).

x.item() should be identical to the previous implementation of numpy.asscalar(x) in every way.

PiperOrigin-RevId: 466197820

Commit:
Automated rollback of commit d20f28464bdd93a66b2f46ffac7cc51c7bac7785

PiperOrigin-RevId: 466217807

Commit:
Continue the job even if linters fail.

Current linter has many false positives and it seems not justify blocking PRs.

Please see also https://github.com/tensorflow/tfx/issues/4434

achoum committed:
Refresh instructions in the .py file for the TensorFlow Decision Forests TFX example.

PiperOrigin-RevId: 466316198

yoonseok312 committed:
Implement Kubernetes Job Runner.

PiperOrigin-RevId: 466356982

tfx-copybara committed:
Renames functions in orchestrator with name *_pipeline_node to *_node

PiperOrigin-RevId: 466380869

tfx-copybara committed:
Upgrade attrs dependency to <22. #159

PiperOrigin-RevId: 466463407

tfx-copybara committed:
Merge pull request #5093 from micahjsmith:check-cache-hits

PiperOrigin-RevId: 466572271

yoonseok312 committed:
Implement rpc for starting pipeline.

PiperOrigin-RevId: 466585150

jeosol committed:
Fix spelling error and improve readability

tfx-copybara committed:
remove status msg from StateRecord

PiperOrigin-RevId: 466843819

tfx-copybara committed:
Added subpipeline task scheduler to experimental orchestrator.

PiperOrigin-RevId: 466949215

tfx-copybara committed:
Add `LatestCreateTime`, a `ResolverOp` that pulls the latest artifact.

PiperOrigin-RevId: 467018357

iindyk committed:
Updating pyarrow dependency for tfx libs to ">=6,<7".

PiperOrigin-RevId: 467044374

tfx-copybara committed:
Add `LatestSpan`, a `ResolverOp` that pulls the artifact with the latest span.

PiperOrigin-RevId: 467055159

tfx-copybara committed:
Add `StaticRange`, a `ResolverOp` that returns artifacts with spans in [start_span, end_span] inclusive.

PiperOrigin-RevId: 467067187

tfx-copybara committed:
Automated rollback of commit 328aaad95a16ed86e4e0f27b698d1c9c2404f66d

PiperOrigin-RevId: 467331399

tfx-copybara committed:
update subpipeline test scheduler test to fix presubmit test failure

PiperOrigin-RevId: 467727456

tfx-copybara committed:
Add back status_msg to avoid potential python json parsing error

PiperOrigin-RevId: 467805898

tfx-copybara committed:
Remove obsolete TODO

PiperOrigin-RevId: 467834056

tfx-copybara committed:
automatically infers the output type of ARTIFACT_LIST in resolver_function

PiperOrigin-RevId: 467979657

tfx-copybara committed:
Explain why overriding `from_json_dict` is necessary for NodeState.

PiperOrigin-RevId: 468061136

yoonseok312 committed:
Implement kubernetes scheduler and orchestrator loop.

PiperOrigin-RevId: 468080576

tfx-copybara committed:
Add execution id to artifact name

PiperOrigin-RevId: 468087642

chongkong committed:
Add __eq__ and __hash__ to resolver_op.Node types.

PiperOrigin-RevId: 468141359

tfx-copybara committed:
update compiler to handle the case that parent_pipeline_context is None

PiperOrigin-RevId: 468220230

tfx-copybara committed:
get individual channels from ResolvedChannel

PiperOrigin-RevId: 468249779

Commit:
Adds a link to Google Python style guide.

PiperOrigin-RevId: 468347484

chongkong committed:
Internal change

PiperOrigin-RevId: 468357707

ConverJens committed:
Merge branch 'tensorflow:master' into enable_env_vars_in_beam_args_through_placeholders

ConverJens committed:
Fixed comments

tfx-copybara committed:
Add garbage collection policies to the DSL and IR

PiperOrigin-RevId: 469491386

rchen152 committed:
Silence some pytype errors.

PiperOrigin-RevId: 469568693

tfx-copybara committed:
Create temporary latest span op

PiperOrigin-RevId: 469778752

tfx-copybara committed:
Create temporary latest create time op

PiperOrigin-RevId: 470171414

tfx-copybara committed:
Merge pull request #5132 from jeosol:master

PiperOrigin-RevId: 470272283

tfx-copybara committed:
Wires additional outputs (match stats + optional confusion) to examplediff tflex executor, and adds the ability to configure confusion analysis.

PiperOrigin-RevId: 470326878

tfx-copybara committed:
Adds an option `keep_all_spans` to `LatestSpan`.

PiperOrigin-RevId: 470780320

tfx-copybara committed:
fix a bug in API UpdatePipeline

PiperOrigin-RevId: 470801341

tfx-copybara committed:
doc update

PiperOrigin-RevId: 470820590

sinopalnikov committed:
Silence type errors generated by new pytype features.

PiperOrigin-RevId: 471209944

tfx-copybara committed:
Add `LatestVersion`, a `ResolverOp` that pulls the artifact with the latest version.

PiperOrigin-RevId: 471564568

chongkong committed:
Add model server logging when infra validation fails.

Also sets TF_CPP_MAX_VLOG_LEVEL=3 (which increases verbosity of the model server log) and fix `tensorflow/serving` version to `2.8.2`.

PiperOrigin-RevId: 471735966

chongkong committed:
Implements compilation routine for new input resolution IR.

PiperOrigin-RevId: 471805874

tfx-copybara committed:
Add `RollingRange` functionality to the `LatestSpans` `ResolverOp`.

PiperOrigin-RevId: 471833509

tfx-copybara committed:
Merge pull request #5124 from jiyongjung0:jiyongjung0-patch-2

PiperOrigin-RevId: 471841186

tfx-copybara committed:
move mlmd_connection_manager to third_party

PiperOrigin-RevId: 471843329

tfx-copybara committed:
use the new API put_lineage_subgraph from MLMD

PiperOrigin-RevId: 471873366

ConverJens committed:
Merge branch 'tensorflow:master' into enable_env_vars_in_beam_args_through_placeholders

Commit:
Fix `tensorflow/serving` version to `2.8.2` temporarily.

PiperOrigin-RevId: 472350308

ConverJens committed:
Introduce a new proto field in executor spec and deprecate the old one

ConverJens committed:
Updated to use the new proto definition

tfx-copybara committed:
Update container entry point to support output parameters from Argo invocation

PiperOrigin-RevId: 472755693

Commit:
Moves typing-extensions dependency from tfx to ml-pipelines-sdk.

PiperOrigin-RevId: 472946455

tfx-copybara committed:
Merge pull request #4981 from tensorflow:sanatmpa1-patch-1

PiperOrigin-RevId: 473245067

Commit:
Upgrades TF dependency to 2.10.

PiperOrigin-RevId: 473454044

yilei committed:
`importlib.abc` is a sub module that needs to be imported explicitly.

PiperOrigin-RevId: 473858436

ConverJens committed:
Updated failing unittests after proto change

ConverJens committed:
Updated failing unittests after proto change

ConverJens committed:
Updated failing unittests after proto change

ConverJens committed:
Updated failing unittests after proto change

ConverJens committed:
Updated proto name after lint error. Updated code to match

tfx-copybara committed:
Implement get_artifacts_to_garbage_collect_for_node

Implements get_artifacts_to_garbage_collect_for_node, which returns output artifacts of the given node to garbage collect.

PiperOrigin-RevId: 474121088

tfx-copybara committed:
set output_data_type in op node

PiperOrigin-RevId: 474166483

Commit:
Splits tflite-support related dependencies to [tflite-support] extra.

The latest tflite-support is not compatible with TF1, so we extract tflite-support related extra to make [examples] compatible with TF1.

PiperOrigin-RevId: 474176052

tfx-copybara committed:
Cap node failure messages. Otherwise node_state property can exceed Spanner cell size limit since it contains all node failure messages of the pipeline run.

Also stop propagating node failure messages to pipeline_status_msg. A list of failed nodes should be enough. Note that pipeline_status_msg also gets added to the node failure message of nodes that have not run.

PiperOrigin-RevId: 474390842

Commit:
Update CAIP Prediction compatibility table to use TF 2.10.

See https://cloud.google.com/ai-platform/prediction/docs/runtime-version-list for the full version list.

PiperOrigin-RevId: 474601840

venkat2469 committed:
TFX 1.10.0 Release

PiperOrigin-RevId: 474620829

tfx-copybara committed:
Set parent and child relationship between pipeline context and pipeline run context

PiperOrigin-RevId: 474659295

iindyk committed:
No op.

PiperOrigin-RevId: 474665002

tfx-copybara committed:
Add initial DistributionValidation TFX OSS component.

PiperOrigin-RevId: 474714837

chongkong committed:
Let `Resolver.inputs` points to the resolved channel instead of the DSL input channels.

Resolver.inputs are only referenced from the compiler code (which is updated in this CL). original input channels are accessible from resolver_node.raw_inputs. We can even remove `raw_inputs` later once we deprecate ResolverConfig compilation.

PiperOrigin-RevId: 474746907

chongkong committed:
Adopt node_inputs_compiler to the compiler.

The new NodeInputs compilation can be enabled by use_input_v2 flag when creating compiler instance. Currently it is False by default, but after verifying the new compilation works without an error, This will be changed to True.

PiperOrigin-RevId: 475194261

chongkong committed:
Fix new IR compiler & input resolution bugs.

I spotted a few bugs during the pipeline end to end tests that I'm fixing
here:

- Resolver node inputs are always optional
- Conditional was evaluating to None due to its InputSpec compiled as hidden
  which was also not visible from the conditional filtering phase. Ideally we
  can add more granularity on the InputSpec visibility, but at the moment,
  making conditional InputSpec non-hidden is the most convenient fix.
- ResolverStrategy config (proto) was compiled to JSON string with schema which
  should be retrieved using data_types_utils.get_parsed_value.
- Added logging for skip signal reason.

PiperOrigin-RevId: 475204333

Commit:
Uses the latest available version when building a container image based on DLVM images.

Also caps google-api-core version to avoid 1.33.0 which is not compatible with tensorboard.

And deleted a workaround for ill-formed TF 2.9 image because we have a new image without the problem.

PiperOrigin-RevId: 475214110

chongkong committed:
Make compiler_test more easy to use.

Changes include:
- now --persist_test_protos directly replaces the source code instead of
  persisting to /tmp.
- No need to specify multiple name; just use a single source of name which is
  a pipeline module
- pbtxt header is auto generated.

PiperOrigin-RevId: 475227472

Commit:
tfx updated to 1.9.1, airflow 2.3.4

rcrowe-google reviewed (approved):
Thanks for simplifying this, it's much better!

mungsoo committed:
Export unexported functions _compute_nodes_to_run and _make_ordered_node_map.

PiperOrigin-RevId: 475395829

chongkong committed:
Added more logging for Launcher and ResolverNodeHandler.

PiperOrigin-RevId: 475469724

Commit:
python 2 lines removed

Commit:
Uses a dedicated proto comparison assertion.

The current test is failing in macos due to a subtle precision difference.

PiperOrigin-RevId: 475557295

Commit:
jupyterlab tfma extension installation removed

Commit:
setup script updated, removed redundant libraries

tfx-copybara committed:
Implement deletion of garbage collected artifacts

PiperOrigin-RevId: 475861882

tfx-copybara committed:
Let _ArtifactRecycler only query pipeline runs from one pipeline

PiperOrigin-RevId: 475872138

tfx-copybara committed:
Statistics gen and Schema gen now crash on empty input examples and statistics respectively.

PiperOrigin-RevId: 475886418

tfx-copybara committed:
Call garbage collection after components are run

Note that garbage collection only supports dry run and does not delete anything yet.

PiperOrigin-RevId: 475954816

singhniraj08 committed:
Update SignatureDefs broken link

SignatureDefs link in Examine your Saved Model section of notebook leads to '404 | Page Not Found error'. Updated link from "../signature_defs" to "https://www.tensorflow.org/tfx/serving/signature_defs".

tfx-copybara committed:
Don't need to update node state to RUNNING if the node state is ready RUNNING

PiperOrigin-RevId: 476175298

tfx-copybara committed:
Add a StopAllPipelines RPC for stopping all active pipelines before shutting down the orchestrator.

PiperOrigin-RevId: 476212209

tfx-copybara committed:
Stop setting artifact.name in TFlex.

PiperOrigin-RevId: 476219968

chongkong committed:
Introduce NullPipelineContext as a parent context of the root pipeline.

This fixes the `node_inputs_compiler` bug for compiling the pipeline begin node of the root pipelines. Pipeline begin node has been compiled as if it is compiling a `Pipeline` as a node of the outer pipeline, which required its outer pipeline context, where root pipeline does not have an outer pipeline nor its context.

Without `NullPipelineContext`, `node_inputs_compiler.compile_node_inputs()` should be changed to take an Optional `PipelineContext` argument, which makes the compiler code much complicated.

PiperOrigin-RevId: 476253221

tfx-copybara committed:
Add an `ExcludeSpans` `ResolverOp`.

PiperOrigin-RevId: 476261712

tfx-copybara committed:
Merge pull request #5247 from tensorflow:singhniraj08-patch-1

PiperOrigin-RevId: 476427910

tfx-copybara committed:
Only set parent-child relationship for newly created pipeline run context

PiperOrigin-RevId: 476491876

sanatmpa1 committed:
Update airflow_workshop.md

Fixing broken link on the file for Airflow CLI

goutham committed:
Conditionally store pipeline IR on disk

PiperOrigin-RevId: 476776870

chongkong committed:
Added Golden IR test for input spec v2 compilation.

PiperOrigin-RevId: 476882336

tfx-copybara committed:
E2E Penguin Kubeflow V2 (Vertex) example pipeline and test

PiperOrigin-RevId: 476885415

tfx-copybara committed:
Sort the outputs of resolver ops LatestCreateTime in ascending orde

PiperOrigin-RevId: 476991504

tfx-copybara committed:
add _implicit_downstream_nodes to NullPipelineContext

PiperOrigin-RevId: 477020674

chongkong committed:
Amend channels docstring.

PiperOrigin-RevId: 477069125

sanatmpa1 committed:
Update tft.md

Fixing the broken link for `Getting Started with TensorFlow Transform`

chongkong committed:
Added output type inferrer decorator to ResolverFunction.

Also renamed type_hint to output_type.

PiperOrigin-RevId: 477135309

tfx-copybara committed:
Refactor `range_config_generator` for readability and downstream use by ExampleGen in `two_step_pipeline_with_dynamic_execution_properties`

PiperOrigin-RevId: 477220680

ConverJens committed:
Fixed issue in step builder

tfx-copybara committed:
Adds canned resolver function `latest_created()`. Also integrates the `MLMDMixins` class testing end-to-end with a test MLMD database.

PiperOrigin-RevId: 477265295

chongkong committed:
Change tf.test.TestCase to test_case_utils.TfxTest.

PiperOrigin-RevId: 477359507

chongkong committed:
Fix pipeline begin node _set_for_each to use parent context.

PiperOrigin-RevId: 477464017

tfx-copybara committed:
Add `ConsecutiveSpansOrSkip ` `ResolverOp`.

PiperOrigin-RevId: 477626604

tfx-copybara committed:
Automated rollback of commit 78cfa243d9da756ddd2597fed27eaf15578e4f98

PiperOrigin-RevId: 477740872

tfx-copybara committed:
Define Node optionality bit.

PiperOrigin-RevId: 477776704

tfx-copybara committed:
Update schema_gen documentation to remove note indicating that schema is generated from the 'train' split.

PiperOrigin-RevId: 477842856

Commit:
Uses a very low threshold to pass the test always.

PiperOrigin-RevId: 477894333

tfx-copybara committed:
Remove some duplicated code in sync_pipeline_task_gen and async_pipeline_task_gen

PiperOrigin-RevId: 477902671

tfx-copybara committed:
Importer will now check that an existing artifact has the same type as the intended output before reusing the existing artifact.

PiperOrigin-RevId: 477933292

tfx-copybara committed:
Refactors `ResolverOp`s to call a `get_valid_artifacts()` utility function to avoid duplicating code.

PiperOrigin-RevId: 478013997

tfx-copybara committed:
Merge pull request #5265 from tensorflow:sanatmpa1-patch-3

PiperOrigin-RevId: 478444260

tfx-copybara committed:
remove unused import in third_party/py/tfx/proto/orchestration/

PiperOrigin-RevId: 478658467

tfx-copybara committed:
Uses a low threshold to pass the test always.

PiperOrigin-RevId: 478676874

tfx-copybara committed:
refactoring, no public text changes

PiperOrigin-RevId: 478702195

tfx-copybara committed:
Only update last_updated_time `if self.state != state`

PiperOrigin-RevId: 478883860

tfx-copybara committed:
Merge pull request #5030 from tensorflow:pindinagesh-patch-3

PiperOrigin-RevId: 478924358

tfx-copybara committed:
Adds external uris in the OutputChannel class.
Adds external_artifact_uris in OutputSpec of IR.
Adds compiler codes and tests to external uris in OutputChannel to external_artifacts_uris in IR.

PiperOrigin-RevId: 478975186

Commit:
Retries TF serving API import if the import fails due to a proto change.

Tensorflow [moved the location of error code definitions](https://github.com/tensorflow/tensorflow/commit/dc2bfae69be58eea63927ffa5900d70b349d1638), but it breaks the code if the version of TF doesn't match with TF serving API.

Because TF serving API doesn't provide nightly packages, we don't have a way to run a test that depends on TF serving API when we need to use TF-nightly.

This code is a hack to make the moved code work. The only symbol that is missing in the new location is '_CODE', so we manually inject it to make the import successful.

PiperOrigin-RevId: 479004136

codesue committed:
Use a copy of beam_pipeline_args in InteractiveContext.run()

Commit:
Deletes obsolete version cap.

Because jax doesn't follow semantic versioning, I chose a bit relaxed upper bound. But our flax example uses 'flax.optim' which is removed at 0.6, so we have to limit flax to 0.6.0.

PiperOrigin-RevId: 479076921

mungsoo committed:
Refactor json_compat relevant functions to a standalone file to solve circular import issuses.

PiperOrigin-RevId: 479094861

tfx-copybara committed:
Get oldest active execution to run for a component.

PiperOrigin-RevId: 479104518

rchen152 committed:
Silence some pytype errors.

PiperOrigin-RevId: 479114598

tfx-copybara committed:
add DistributionValidator facade TFlex component

PiperOrigin-RevId: 479130128

tfx-copybara committed:
Add more details in the error message to help user figure out what are the node
names when trying to do partial runs.

PiperOrigin-RevId: 479138857

tfx-copybara committed:
Handle missing split pairs in the ExampleDiff and DistributionValidator components.

PiperOrigin-RevId: 479162286

Commit:
Introduce an upper cap for google-cloud-aiplatform.

The latest version (1.18) has a conflict on protobuf version with tensorflow.

PiperOrigin-RevId: 479168715

tfx-copybara committed:
Handle __enter__ for Metadata handler in MLMDConnectionManager and reset reader handle dictionary in __exit__.

PiperOrigin-RevId: 479201456

tfx-copybara committed:
refactor only

PiperOrigin-RevId: 479217511

tfx-copybara committed:
Importer will now use the most recently created artifact when reusing an existing artifact instead of the one with the highest ID.

PiperOrigin-RevId: 479241179

chongkong committed:
Automated rollback of commit 75b2101917768711f27a06a012d37c58254a2182

PiperOrigin-RevId: 479244343

Commit:
Announces the deprecation of TF1 support.

The long term support of TensorFlow Enterprise 1.15 ends at [2022-10-30](https://cloud.google.com/tensorflow-enterprise/docs/overview#long_term_version_support). We will release one more version with TF1 support and then stop supporting it.

PiperOrigin-RevId: 479272824

tfx-copybara committed:
Fix gather_file_descriptors() with topological sort

PiperOrigin-RevId: 479320008

tfx-copybara committed:
Add a `SlidingWindow` `ResolverOp`.

PiperOrigin-RevId: 479339583

tfx-copybara committed:
Add a `static_range` resolver function corresponding to `StaticRange` in TFX.

PiperOrigin-RevId: 479342528

tfx-copybara committed:
Internal BUILD change

PiperOrigin-RevId: 479372754

tfx-copybara committed:
Minor cleanup in `ops_utils.py`, addressing goutham@'s comments in cl/478013997.

PiperOrigin-RevId: 479381075

Commit:
Splits jax/flax dependencies into a separate [flax] extra.

PiperOrigin-RevId: 479503475

tfx-copybara committed:
Add a handle_stop method to base executor operator

PiperOrigin-RevId: 479619298

goutham committed:
Gracefully handle internal errors during orchestration

Attempt to terminate the concerned pipeline instead of letting the exception raise and crash the orchestrator.

PiperOrigin-RevId: 479678466

tfx-copybara committed:
Update orchestrator to handle a list of Dict from a node

PiperOrigin-RevId: 479848629

gnecula committed:
Update the penguin example with flax_experimental to use optax.

optax is now the supported optimizer libary for Flax, the old flax.optim is deprecated.

Followed upgrade recipe at https://flax.readthedocs.io/en/latest/advanced_topics/optax_update_guide.html

PiperOrigin-RevId: 480086295

tfx-copybara committed:
Add a `SkipIfEmptyOpTest` `ResolverOp`.

Corresponds to `min_spans` in the TFX `RangeConfig` proto. This `ResolverOp` will also be needed in the `sequential_rolling_range` resolver function.

PiperOrigin-RevId: 480135046

tfx-copybara committed:
Sort execution by time if the executions do not have _EXTERNAL_EXECUTION_INDEX

PiperOrigin-RevId: 480150989

Commit:
PR #5031: udf_utils: fix python path modification

Please approve this CL. It will be submitted automatically, and its GitHub pull request will be marked as merged.

Imported from GitHub PR #5031

This should fix an issue where in some cases `sys.path` and (especially) `$PYTHONPATH` is not correctly restored after being temporarily modified. This was the cause of problems in subsequent code (e.g. wrong module order, etc.).

Copybara import of the project:

  - 96ebe6d318d7c064bdd0059b74565e722b0e1faa udf_utils: fix python path modification by japm48 <japm48gh@gmail.com>
  - d96ff7eb4957acfde003f82e249aec7afe221a86 udf_utils: pylint fixes by japm48 <japm48gh@gmail.com>
  - 416c52041b3a8b40787788c5622aa25c72d3f1b2 Merge d96ff7eb4957acfde003f82e249aec7afe221a86 into 14c15... by japm48 <japm48gh@gmail.com>

COPYBARA_INTEGRATE_REVIEW=https://github.com/tensorflow/tfx/pull/5031 from japm48:patch-1 d96ff7eb4957acfde003f82e249aec7afe221a86
PiperOrigin-RevId: 480229518

tfx-copybara committed:
Introduces a mechanism to the Artifact class to add implicit properties to all Artifact subclasses.

PiperOrigin-RevId: 480239186

tfx-copybara committed:
If a failed execution has returned an executor_output, publish the executor_output.execution_result to MLMD.

PiperOrigin-RevId: 480247598

goutham committed:
Support skipping execution of nodes

PiperOrigin-RevId: 480262736

goutham committed:
Fix: use HasField for proto fields

PiperOrigin-RevId: 480398860

tfx-copybara committed:
Merge pull request #5308 from codesue:codesue/5126-use-list-copy-in-IC-run

PiperOrigin-RevId: 480531396

tfx-copybara committed:
Add keep_recent_spans_and_versions to GC policies

PiperOrigin-RevId: 480563247

tfx-copybara committed:
Adds support for Artifact/Channel properties to use the upcoming MLMD PROTO property type. This new property type is presently disabled by default until it is fully supported by MLMD.

PiperOrigin-RevId: 480752563

tfx-copybara committed:
TFleX: Optimize the number of MLMD queries in garbage collection

PiperOrigin-RevId: 480777577

Commit:
Makes a copy of the job state to avoid changes between comparisons.

PiperOrigin-RevId: 480781188

tfx-copybara committed:
Add a `rolling_range` resolver function corresponding to `RollingRange` in TFX.

PiperOrigin-RevId: 480805720

tfx-copybara committed:
Update TFT examples to get DatasetMetadata from the tft module instead of dataset_metadata.

PiperOrigin-RevId: 480875826

tfx-copybara committed:
refactor publishing execution result

PiperOrigin-RevId: 480961029

tfx-copybara committed:
Merge pull request #4861 from ConverJens:enable_env_vars_in_beam_args_through_placeholders

PiperOrigin-RevId: 481132090

tfx-copybara committed:
Fix typo in TFT docs.

PiperOrigin-RevId: 481150313

tfx-copybara committed:
Merge pull request #5258 from tensorflow:sanatmpa1-patch-1

PiperOrigin-RevId: 481171799

goutham committed:
Changes to call some functions post MLMD commit

PiperOrigin-RevId: 481202375

tfx-copybara committed:
Introduces merge_utils module for merging objects during pipeline orchestration. This change adds a method called "merge_output_artifacts", which will merge a pre-execution version of an output artifact defined by the orchestrator with it's post-execution version returned by the component.

PiperOrigin-RevId: 481534042

tfx-copybara committed:
Add ProtoSerializationFormat.INLINE_FILE_TEXT_FORMAT

PiperOrigin-RevId: 481623743

tfx-copybara committed:
assume single key from prediction_log request inputs instead of assuming single key with fixed name: "examples"

PiperOrigin-RevId: 481707063

chongkong committed:
Add `unwrap_dict_key` to the @resolver_function decorator.

PiperOrigin-RevId: 481799346

goutham committed:
Store pipeline execution mode as property for quick retrieval without parsing IR

PiperOrigin-RevId: 481809451

chongkong committed:
Rename mlmd_handler -> mlmd_handle.

PiperOrigin-RevId: 481881752

tfx-copybara committed:
Refactors artifact Property and PropertyType classes out of artifact.py into a new module, artifact_property.py.

PiperOrigin-RevId: 481908917

tfx-copybara committed:
Handles input resolution failure as component failure in the experimental orchestrator.

PiperOrigin-RevId: 482024744

singhniraj08 committed:
Update 404 broken link

updated tfdv.visualize_statistics() broken 404 link 'tfdv.visualize_statistics' to 'https://www.tensorflow.org/tfx/data_validation/api_docs/python/tfdv/visualize_statistics'

chongkong committed:
Add `name` parameter for test_case_utils.MlmdMixins put methods.

PiperOrigin-RevId: 482124424

tfx-copybara committed:
Adds a new implicit property, 'is_external'.
Modifies the tests to add 'is_external' in artifacts.

PiperOrigin-RevId: 482136143

tfx-copybara committed:
Merge pull request #5391 from tensorflow:singhniraj08-patch-3

PiperOrigin-RevId: 482201029

tfx-copybara committed:
refactor

PiperOrigin-RevId: 482284696

tfx-copybara committed:
TFleX GC: Bugfix for URI deletion, for both file and directory

PiperOrigin-RevId: 482355730

tfx-copybara committed:
Automated rollback of commit 294c74a4a6dc2ac369c92ff6779c6fb9d4cc3986

PiperOrigin-RevId: 482359450

tfx-copybara committed:
Automated rollback of commit 197a9eff35f2513592ca6c20366e1272c2bdcb9f

PiperOrigin-RevId: 482366663

tfx-copybara committed:
Adds a boolean "is_external" attribute to the Artifact class.

PiperOrigin-RevId: 482369085

singhniraj08 committed:
User Guide Fix: Update Broken Link

updated broken `training/serving skew` link from '#training-serving-skew-detection' to 'https://www.tensorflow.org/tfx/guide/tfdv#training-serving_skew_detection'

tfx-copybara committed:
Merge pull request #5403 from tensorflow:singhniraj08-patch-4

PiperOrigin-RevId: 482493353

tfx-copybara committed:
Fixes encode error when concatenating a placeholder with itself.

PiperOrigin-RevId: 482495709

tfx-copybara committed:
Fix or ignore some pytype errors.

PiperOrigin-RevId: 482622416

tfx-copybara committed:
GC: Filters out external artifacts

PiperOrigin-RevId: 482683230

tfx-copybara committed:
Removes the public Placeholder.from_comparison class method and replaces its handful of internal usages with the unwrapped equivalent.

PiperOrigin-RevId: 482685055

goutham committed:
Make PipelineView use pipeline_id instead of orchestrator internal pipeline_uid

PiperOrigin-RevId: 482690012

tfx-copybara committed:
Generates external output artifacts by predefined_artifact_uris in OutputSpec of IR.
Makes orchestrator omits lifecycle management for external artifacts.

PiperOrigin-RevId: 482695196

chongkong committed:
Add exc_info=True for input resolution error logging.

This argument is needed for traceback logging, and also properly shows the chained exceptions.

PiperOrigin-RevId: 482714080

chongkong committed:
Fix OutputSpec IR compilation for OutputChannel specific fields.

Previously it was assuming the argument is a `Channel`, but it is actually a `BaseChannel` (we call it with pipeline.inputs when compiling a PipelineBegin node) so added a branch guard. This fixes the bug that the subpipeline can only take the `OutputChannel`.

Background:

`OutputChannel`, or a `component.outputs["x"]` is not only the possible inputs to another components (i.e. is a `BaseChannel`) but also the place to configure output-related configuration such as `additional_(custom_)properties`, `garbage_collection_policy`, `external_artifact_uris`, etc.

`OutputSpec` is the place to store those output-only information. However there is a case we also accept non-`OutputChannel`; currently only when compiling a PipelineBegin node, which can technically take any `BaseChannel` raw inputs.

Still every node IR needs the most basic `OutputSpec` IR which only stores the output key -> artifact type mapping. They're currently used from the UI to render the pipeline graph.

PiperOrigin-RevId: 482721091

tfx-copybara committed:
Change all new executions to CANCEL when calling API StopNode

PiperOrigin-RevId: 482802752

tfx-copybara committed:
Set is_external = True for artifacts produced by the Importer node.

PiperOrigin-RevId: 483254490

tfx-copybara committed:
GC: Add dry_run to monitor the payload deletion

PiperOrigin-RevId: 483369519

tfx-copybara committed:
Move TriggerStrategy dsl to NodeExecutionOptions

PiperOrigin-RevId: 483493042

chongkong committed:
Make new NodeInputs compilation routine the default

PiperOrigin-RevId: 483568026

tfx-copybara committed:
Updates Placeholder operators to always return a new copy of Placeholder object, instead of mutating it.

PiperOrigin-RevId: 483681508

tfx-copybara committed:
Implement experimental node success optionality

PiperOrigin-RevId: 483707808

goutham committed:
Opt-in capability for orchestrating concurrent pipeline runs

PiperOrigin-RevId: 483736070

goutham committed:
Restrict async pipelines when concurrent pipeline runs are enabled

PiperOrigin-RevId: 483843131

tfx-copybara committed:
GC: Actually delete the payload

PiperOrigin-RevId: 483896630

tfx-copybara committed:
GC: Bugfix of _try_delete_uri when doing GC twice with the same URI

PiperOrigin-RevId: 483940548

tfx-copybara committed:
Support concatenating placeholders with itself.

The previous fix using deepcopy would translate `a + '-' + a` into `(a + '-' + a + '-')` as the first concatenation would be included in the deepcopy operation. Instead, we introduce a specific method `concat_placeholders` which is used for concatenating a list of placeholders. At the same time, we disallow concat operations for a placeholder with itself to avoid the problem mentioned previously.

PiperOrigin-RevId: 484086795

Commit:
Splits test files for sequence examples.

PiperOrigin-RevId: 484095887

rcrowe-google committed:
Adding recommenders and ranking tutorials

PiperOrigin-RevId: 484109736

tfx-copybara committed:
no op.

PiperOrigin-RevId: 484122099

tfx-copybara committed:
Do not attempt to remove Importer's output_artifacts upon execution failures. They are external URIs.

PiperOrigin-RevId: 484133020

tfx-copybara committed:
Latest pipeline run op

PiperOrigin-RevId: 484160161

chongkong committed:
Fix KubeFlow entrypoint bug that fails on resolver node post-execution logging.

When Resolver node outputs is used as an inputs, the component IR no longer contains `input_spec.channels`, but the logging was still referencing it for the artifact type retrieval. If this artifact type information is really important we can consider adding a dedicated field to the `InputSpec`. Note that `InputSpec` <-> `BaseChannel` <-> `ArtifactType` has 1:1:1 correspondence.

PiperOrigin-RevId: 484169320

tfx-copybara committed:
Add an `AllSpans` `ResolverOp` and an `all_spans()` resolver function.

Returns the sorted artifacts with unique spans.

PiperOrigin-RevId: 484269209

tfx-copybara committed:
Add `min_spans` for `static_range()` and `rolling_range()` resolver functions.

PiperOrigin-RevId: 484285890

tfx-copybara committed:
update run_resolver_op in test_utils.py so that it can set context to ResolverOp

PiperOrigin-RevId: 484300999

tfx-copybara committed:
Add a `sequential_rolling_range` resolver function corresponding to `SequentialRollingRange` in TFX.

PiperOrigin-RevId: 484391496

tfx-copybara committed:
GC: Simplify artifact's state transition; directly delete the payload

PiperOrigin-RevId: 484411175

tfx-copybara committed:
Remove compiler parameter from compiler_test.py

PiperOrigin-RevId: 484458754

chsigg committed:
Use @bazel_skylib's bazel version check instead of implementing our own.

PiperOrigin-RevId: 484471344

tfx-copybara committed:
Distinguish execution states of failed and canceled

PiperOrigin-RevId: 484647656

genehwung committed:
noop

PiperOrigin-RevId: 484688926

dev-nid committed:
Merge branch 'master' into upgrading-pipeline-codes-and-utils

Commit:
Instructions updated, images updated

rcrowe-google commented:
Since this is Prianka's repo I'll let her merge this when it's ready.

Commit:
install packages in a single command

Commit:
DAG images updated, Vertex AI instance creation instructions updated

Commit:
Permission changed for setup_demo.sh
